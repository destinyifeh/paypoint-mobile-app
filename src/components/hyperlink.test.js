'use strict';

import React from 'react';
import { Text } from 'react-native';
import renderer from 'react-test-renderer';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Hyperlink from './hyperlink';

jest.mock('react-navigation', () => ({ withNavigation: component => component}));

const mockProps = {
  navigation: {
    navigate: jest.fn(),
    replace: jest.fn()
  },
  loginSuccess: jest.fn(),
  isSubmitting: true
}

configure({
  adapter: new Adapter()
});

it('renders correctly', () => {
  const tree = renderer
    .create(<Hyperlink />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('onPressOut functions as expected', () => {
  const onPressEvent = jest.fn();
  onPressEvent.mockReturnValue('Link on press invoked');
  const wrapper = shallow(<Hyperlink
    {...mockProps}
    onPress={ onPressEvent } 
  />);
  wrapper.find(Text).first().props().onPress();
  expect(onPressEvent.mock.calls.length).toBe(1);
})

it('replace as false functions as expected', () => {
  const onPressEvent = jest.fn();
  onPressEvent.mockReturnValue('Link on press invoked');
  const wrapper = shallow(<Hyperlink
    {...mockProps}
  />);
  wrapper.find(Text).first().props().onPress();
  const instance = wrapper.instance();

  expect(instance.hasNavigated).toBe(true);
})

it('replace as true functions as expected', () => {
  const onPressEvent = jest.fn();
  onPressEvent.mockReturnValue('Link on press invoked');
  const wrapper = shallow(<Hyperlink
    replace
    {...mockProps}
  />);
  wrapper.find(Text).first().props().onPress();
  const instance = wrapper.instance();

  expect(instance.hasReplaced).toBe(true);
})
