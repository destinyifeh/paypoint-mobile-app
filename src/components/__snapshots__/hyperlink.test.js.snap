// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Text
  onPress={[Function]}
  style={
    Object {
      "color": "#00B8DE",
    }
  }
/>
`;
